finagle-redis/src/main/scala/com/twitter/finagle/Redis.scala

```diff
@@ -63,4 +66,48 @@ object Redis extends Client[Command, Reply] {
 
   def newService(dest: Name, label: String): Service[Command, Reply] =
     client.newService(dest, label)
+
+  object Subscribe extends com.twitter.finagle.Client[SubscribeCommand, Unit] {
+
+    object SubscribeClient {
+      /**
+       * Default stack parameters used for redis client.
+       */
+      val defaultParams: Stack.Params = StackClient.defaultParams +
+        param.ProtocolLibrary("redis.subscribe")
+
+      /**
+       * A default client stack which supports the pipelined redis client.
+       */
+      def newStack: Stack[ServiceFactory[SubscribeCommand, Unit]] = StackClient.newStack
+        .replace(DefaultPool.Role, SingletonPool.module[SubscribeCommand, Unit])
+    }
+
+    case class SubscribeClient(
```

Comment:
Subscription is more complicated than transactions. Consider the following scenario:
   
    send: subscribe foo      <-- The client subscribes to a channel,

    recv: subscribe foo 1    <-- and receives an acknowledgement like this. The
                                 number at the end is the number of channels the
                                 client subscribed to. The client will exit from
                                 pub/sub mode, when it drops down to 0.

    send: unsubscribe foo    <-- Not interested in this channel any more.

    send: subscribe bar      <-- Another subscription, before the acknowledgement
                                 message arrives.

    recv: unsubscribe foo 0  <-- It's not safe to reuse this connection. :(

    recv: subscribe bar 1